gui/esp.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <vector>
+#include <mutex>
+
+#include "state.hpp"
+
+#include <../includes/imgui/imgui.h>
+
+static bool IsWithinScreenBounds(Vector2& pos)
+{
+	return pos.x < (float)Screen_get_width(nullptr) && pos.y < (float)Screen_get_height(nullptr);
+}
+
+static ImVec2 WorldToScreen(Vector2 pos)
+{
+	const ImVec2 winsize = { (float)Screen_get_width(nullptr), (float)Screen_get_height(nullptr) };
+
+	// These values are not random, i found them by random testing
+	float scale = winsize.y / 1080.0f;
+	float view = (180 * scale) / (State.CameraHeight / 3); // The value 180 is specific for 1080p so we need to scale it for other resolutions
+	// NOTE : We always need to divide the camera height by 3
+	// since 3 is the default zoom in the menu for some reason
+
+	PlayerControl* localPlayer = PlayerControl__TypeInfo->static_fields->LocalPlayer;
+	Vector2 localPos = PlayerControl_GetTruePosition(localPlayer, nullptr);
+
+	/*Camera->GetMain() function*/
+	auto mainCamera = Camera_get_main(nullptr);
+	Transform* cameraTransform = Component_get_transform(reinterpret_cast<Component*>(mainCamera), nullptr);
+	Vector3 cameraPosition = Transform_get_position(cameraTransform, nullptr);
+
+	/*Calculation to compensate for Camera movement*/
+	cameraPosition.x = localPos.x - (localPos.x - cameraPosition.x);
+	cameraPosition.y = localPos.y - (localPos.y - cameraPosition.y);
+
+	// Here we transform the world position to the screen position
```

Comment:
can't we use the worldtoscreen function by the game here ?

Reply:
This will mess up the scaling of the positions, i am not shure if this can easily be done with the unity solution. If we would use the unity solution then scaling the character bounds would have to find the bounding boxes of the characters (bounding boxes could also be bigger than the player and ruin the esp).